src/main/java/eu/europeana/metis/sandbox/domain/Event.java

```diff
@@ -0,0 +1,59 @@
+package eu.europeana.metis.sandbox.domain;
+
+import static java.util.Objects.requireNonNull;
+
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.common.Step;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+public class Event<T extends Record> {
+
+  private final T body;
+  private final Status status;
+  private final Step step;
+
+  private final String exception;
+
+  public Event(T body, Step step) {
+    requireNonNull(body, "Body must not be null");
+    requireNonNull(step, "Step must not be null");
+    this.status = Status.SUCCESS;
+    this.body = body;
+    this.exception = null;
+    this.step = step;
+  }
+
+  public Event(T body, Step step, Exception exception) {
+    requireNonNull(body, "Body must not be null");
+    requireNonNull(step, "Step must not be null");
+    this.status = exception == null ? Status.SUCCESS : Status.FAIL;
+    this.body = body;
+    this.exception = exception == null ? null : ExceptionUtils.getStackTrace(exception);
+    this.step = step;
+  }
+
+  public Event(T body, Step step, String exception) {
+    requireNonNull(body, "Body must not be null");
+    requireNonNull(step, "Step must not be null");
+    this.status = exception == null ? Status.SUCCESS : Status.FAIL;
+    this.body = body;
+    this.exception = exception;
+    this.step = step;
+  }
```

Comment:
It's better to let two of these constructors call the third directly, instead of having three blocks that do more or less the same.

Reply:
Yes I thought of that, but resolution was not working, anyways I already change this code in the next review I have pending to send